Wobble/Scheduling/TaskHandler.cs

```diff
@@ -71,42 +82,57 @@ public CancellationToken Run(T input, int delay = 0)
                 Source.Cancel();
                 Source.Dispose();
                 Source = new CancellationTokenSource();
-
-                IsCompleted = false;
-                IsCancelled = false;
-                IsRunning = true;
+                TaskRun(Source.Token, input, delay);
```

Comment:
How are you calling an async function from a regular function?

Reply:
I don't know.

Reply:
So from what I understand from reading the C# reference, the way this works is that the code runs up until the first `await` that actually suspends the execution (rather than completing instantly), then the control returns to the calling point and resumes from there, and when the expression that suspended the `await` completes the execution is resumed there on a separate thread or something.